src/Symfony/Bundle/FrameworkBundle/Command/ServerCommand.php

```diff
@@ -41,4 +41,25 @@ protected function getLockFile($address)
     {
         return sys_get_temp_dir().'/'.strtr($address, '.:', '--').'.pid';
     }
+
+    protected function isOtherServerProcessRunning($address)
+    {
+        $lockFile = $this->getLockFile($address);
+
+        if (file_exists($lockFile)) {
+            return true;
+        }
+
+        list($hostname, $port) = explode(':', $address);
```

Comment:
If i use the `parse_url`, the code wil be a bit more verbose to solve some bugs if the user enters a bad address. ex:
```
// print_r(parse_url('127.0.0.1:8000'));
Array
(
    [host] => 127.0.0.1
    [port] => 8000
)
```
```
// print_r(parse_url('127.0.0.1:'));
Array
(
    [scheme] => 127.0.0.1
)
```
```
// print_r(parse_url(':800'));
NULL
```